pandas/_typing.py

```diff
@@ -18,7 +18,7 @@
 AnyArrayLike = TypeVar("AnyArrayLike", "ExtensionArray", "Index", "Series", np.ndarray)
 ArrayLike = TypeVar("ArrayLike", "ExtensionArray", np.ndarray)
 DatetimeLikeScalar = TypeVar("DatetimeLikeScalar", "Period", "Timestamp", "Timedelta")
-Dtype = Union[str, np.dtype, "ExtensionDtype"]
+Dtype = Type[Union[str, float, bool, np.dtype, "ExtensionDtype"]]
```

Comment:
```suggestion
Dtype = Union[ExtensionDtype, Type[Union[str, float, int, complex, bool]]]
```

Somewhat more complicated but the `ExtensionDtype` needs to be an instance, so should exclude from the `Type` element.

To illustrate:

```python
>>> pd.Series(range(1), dtype=pd.Int64Dtype)  # Not what you would expect
0    0
dtype: object
>>> pd.Series(range(1), dtype=pd.Int64Dtype())  # This instance works though
0    0
dtype: object
>>> pd.Series(range(1), dtype=int) # contrast to the builtin, which uses Type[int]
0    0
dtype: int64
```

cc @TomAugspurger in case there's something I'm missing with the ExtensionDtype

Also @simonjayhawkins do you mind if we include `int` in the Union? Understood that it gets replaced as part of the mypy type hierarchy, but given ``dtype=int`` is a valid construct it serves as literal documentation and requires less knowledge of how mypy works. Whenever we get to Python 3.8 as the min version these *should* be converted to `Literal` annotations anyway

Reply:
```
>>> pd.Series(range(1), dtype=pd.Int64Dtype)
```

should probably raise. I don't think we can automatically make that work since we have parametrized dtypes. Not sure what that does to the type hint.

Reply:
> Also @simonjayhawkins do you mind if we include `int` in the Union?

no problem.

Reply:
👋 Sorry for the silence on my end. I'm picking this up again now. I didn't follow your comment, @WillAyd (though I committed your suggestion - hence the `Outdated` flag). Does this requested review still apply? Thanks.